app/tests/archives_tests/test_views.py

```diff
@@ -1259,7 +1264,7 @@ def get_resp():
     "viewname", ("archives:item-job-list", "archives:item-detail")
 )
 def test_archive_item_permissions_detail_and_list(viewname, client):
-    archive = ArchiveFactory()
+    archive = ArchiveFactory(workstation=WorkstationFactory())
```

Comment:
The view should work without a workstation being set.

Reply:
I got an error when creating the session url (for the civ details view), so when calling `workstation.slug`, workstation is None

Reply:
After this fix, the test worked

Reply:
Nowhere in this test is `archive.workstation` used. It should not be necessary.

Reply:
I tried again without workstation and I got this error (which is the same one I had before)

<img width="1678" alt="Screenshot 2024-09-19 at 16 57 23" src="https://github.com/user-attachments/assets/fe5a5628-c232-4631-904f-cb4242e3635c">

The test calls `get_view_for_user` for viewname `archives:item-detail`, which in turn calls `ArchiveItemDetailView` which uses the `template archives/archive_item_detail.html` which inherits `components/civ_set_detail.htm` which calls this line ([here](https://github.com/comic/grand-challenge.org/blob/9584c6b8217bec7225f6c0c1f994dd0a81dc05c7/app/grandchallenge/components/templates/components/civ_set_detail.html#L34)):

`{% workstation_session_control_data workstation=object.base_object.workstation context_object=object archive_item=object config=object.base_object.workstation_config %} `

which calls the `slug` of `workstation` [here](https://github.com/comic/grand-challenge.org/blob/9584c6b8217bec7225f6c0c1f994dd0a81dc05c7/app/grandchallenge/workstations/templatetags/workstations.py#L84)

Am I wrong in tracing this issue?

Reply:
Think of the fundamental problem. A workstation is not required for an archive. You have a bit of code that is assuming that a workstation is required for an archive. You need to fix that code rather than changing the requirement so that all archives must define a workstation.